typescript-requiring-type-checking.js

```diff
@@ -21,6 +21,37 @@ module.exports = {
         */
         '@typescript-eslint/prefer-regexp-exec': 'off',
 
+        /*
+            This rule can be confusing if you're not familiar with JavaScript's rules for binding `this`,
+            but in our experience when it reports an error it is usually indicitave of a real bug.
+
+            Common ways to resolve this error include:
+            1. Wrapping the function with an arrow function, which establishes `this` based on the context
+               the function is defined in rather than where it's called from.
+               ```
+                    // Good, this will refer to the class where this code is defined
+                    array.sort(() => this.compare());
+
+                    // Bad, this will refer to the context where this code is invoked from
+                    array.sort(this.compare);
+                ```
+            2. For callbacks passed to `addEventListener` which need to be referenced in `removeEventListener`,
+               the arrow function should be stored in a variable since it must refer to the same instance of the function.
+               ```
+                    // Good
+                    const handler = event => this.clickHandler();
+                    elt.addEventListener('click', handler);
+                    elt.removeEventListener('click', handler);
+
+                    // Bad, violates this rule
+                    elt.addEventListener('click', this.clickHandler);
+                    elt.removeEventListener('click', this.clickHandler);
+
+                    // Bad, removeEventListener isn't removing the original handler
+                    elt.addEventListener('click', event => this.clickHandler());
+                    elt.removeEventListener('click', event => this.clickHandler());
+            3. Changing the function to be static if it doesn't need to refer to the `this` instance.
```

Comment:
I don't think this fixes the issue, because static methods can just as well use `this` to refer to the class and call other static methods.

Reply:
True. My understanding is that `ignoreStatic` will cause the rule not to fire for static methods, which would be correct if the static method doesn't use `this`. I tried re-wording it to express this; let me know what you think.

Reply:
I guess with respect to how the rule is configured that is correct. However, there is nothing enforcing static methods not use `this`. Someone could come along later and one. Maybe that is a better discussion around the rule's configuration rather than what your intentions are with this PR.